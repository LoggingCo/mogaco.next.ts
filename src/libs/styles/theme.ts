import { DefaultTheme } from 'styled-components';

const palette = {
  primary: {
    100: '#BFDEEE',
    200: '#9DCDE7',
    300: '#3AA5DD',
  },
  subColor: '#EBF1F4',
  fontColor: '#222222',
  fontSubColor: '#ffffff',
  error: '#EB5757',
  subError: '#FFBA52',
  gray: {
    100: '#e9e9e9',
    200: '#D9D9D9',
    300: '#656565',
  },
  yellow: '#FFE381',
  puple: '#EDC4FC',
  sky: '#A4EEFF',
} as const;

const fontSize = {
  small: '14px',
  medium: '16px',
  large: '18px',
  xLarge: '24px',
  xxLarge: '32px',
} as const;

const fontWeight = {
  thin: '100',
  regular: '400',
  medium: '500',
  bold: '700',
} as const;

const lineHeight = {
  small: '16px',
  medium: '20px',
  large: '24px',
  xLarge: '28px',
} as const;

export type PaletteTypes = typeof palette;
export type PaletteKeyTypes = keyof typeof palette;
export type FontSizeTypes = typeof fontSize;
export type FontWeightTypes = typeof fontWeight;
export type LineHeightTypes = typeof lineHeight;

const theme: DefaultTheme = {
  palette,
  fontSize,
  fontWeight,
  lineHeight,
};

export default theme;
