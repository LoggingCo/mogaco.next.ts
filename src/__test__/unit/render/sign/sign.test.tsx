import { render, screen, waitFor } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '@/libs/styles/theme';
import userEvent from '@testing-library/user-event';
import Sign from '@/pages/sign';

describe('<Sign />', () => {
  it('로그인/회원가입 페이지가 렌더링 되고 스냅샷을 비교한다', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Sign />
      </ThemeProvider>,
    );
    const $loginButton = screen.getByRole('button', { name: '로그인' });

    expect($loginButton).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('아이디와 비밀번호를 입력해야만 버튼이 활성화된다', () => {
    render(
      <ThemeProvider theme={theme}>
        <Sign />
      </ThemeProvider>,
    );

    const $emailInput = screen.getByPlaceholderText('이메일 입력하세요');
    const $passwordInput = screen.getByPlaceholderText('비밀번호를 입력하세요');
    const $loginButton = screen.getByRole('button');

    userEvent.type($emailInput, 'test');
    userEvent.type($passwordInput, 'test');
    userEvent.click($loginButton);

    expect($loginButton).toBeEnabled();
  });

  it('회원가입 버튼 클릭 시 로그인 폼이 아닌 회원 가입이 보여야한다', async () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Sign />
      </ThemeProvider>,
    );

    const $signUpList = screen.getByRole('sign-up');
    userEvent.click($signUpList);

    await waitFor(() => {
      const $signUpBtn = screen.getByRole('button', { name: '회원가입' });
      expect($signUpBtn).toBeInTheDocument();
    });
  });
});
