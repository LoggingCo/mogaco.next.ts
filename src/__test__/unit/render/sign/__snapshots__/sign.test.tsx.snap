// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> 로그인/회원가입 페이지가 렌더링 되고 스냅샷을 비교한다 1`] = `
<div>
  <div
    class="sc-kDvujY cwVVHS"
  >
    <div
      class="sc-ipEyDJ hflXJS"
    >
      <div
        class="sc-dkrFOg jcNiDM"
      >
        <a
          href="/"
        >
          mogaco
        </a>
      </div>
      <div
        class="sc-eDWCr cGxhHB"
      >
        오늘도
        <br />
        모가코,
      </div>
    </div>
    <div
      class="sc-bqWxrE iEntZY"
    >
      <div
        class="sc-ksBlkl RdDIT"
      >
        <ul>
          <li
            role="sign-in"
          >
            로그인
          </li>
          <li
            role="sign-up"
          >
            회원가입
          </li>
        </ul>
      </div>
      <div
        class="sc-hLBbgP FbSkc"
      >
        <form
          class="sc-eDvSVe juBFCv"
        >
          <div
            class="sc-jSUZER jhgDYF"
          >
            <input
              name="email"
              placeholder="이메일 입력하세요"
            />
            <span>
              이메일
            </span>
          </div>
          <div
            class="sc-jSUZER jhgDYF"
          >
            <input
              name="password"
              placeholder="비밀번호를 입력하세요"
              type="password"
            />
            <span>
              비밀번호
            </span>
          </div>
          <button>
            로그인
          </button>
          <div
            class="sc-gKPRtg eEOnMt"
          >
            아직 모가코 회원이 아니닌가요? 
            <span>
              모가코와 함께하기
            </span>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Home /> 회원가입 버튼 클릭 시 로그인 폼이 아닌 회원 가입이 보여야한다 1`] = `
<div>
  <div
    class="sc-kDvujY cwVVHS"
  >
    <div
      class="sc-ipEyDJ hflXJS"
    >
      <div
        class="sc-dkrFOg jcNiDM"
      >
        <a
          href="/"
        >
          mogaco
        </a>
      </div>
      <div
        class="sc-eDWCr cGxhHB"
      >
        오늘도
        <br />
        모가코,
      </div>
    </div>
    <div
      class="sc-bqWxrE iEntZY"
    >
      <div
        class="sc-ksBlkl RdDIT"
      >
        <ul>
          <li
            role="sign-in"
          >
            로그인
          </li>
          <li
            role="sign-up"
          >
            회원가입
          </li>
        </ul>
      </div>
      <div
        class="sc-hLBbgP FbSkc"
      >
        <form
          class="sc-eDvSVe juBFCv"
        >
          <div
            class="sc-jSUZER jhgDYF"
          >
            <input
              name="email"
              placeholder="이메일 입력하세요"
            />
            <span>
              이메일
            </span>
          </div>
          <div
            class="sc-jSUZER jhgDYF"
          >
            <input
              name="password"
              placeholder="비밀번호를 입력하세요"
              type="password"
            />
            <span>
              비밀번호
            </span>
          </div>
          <button>
            로그인
          </button>
          <div
            class="sc-gKPRtg eEOnMt"
          >
            아직 모가코 회원이 아니닌가요? 
            <span>
              모가코와 함께하기
            </span>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Sign /> 로그인/회원가입 페이지가 렌더링 되고 스냅샷을 비교한다 1`] = `
<div>
  <div
    class="sc-kDvujY cwVVHS"
  >
    <div
      class="sc-ipEyDJ hflXJS"
    >
      <div
        class="sc-dkrFOg jcNiDM"
      >
        <a
          href="/"
        >
          mogaco
        </a>
      </div>
      <div
        class="sc-eDWCr cGxhHB"
      >
        오늘도
        <br />
        모가코,
      </div>
    </div>
    <div
      class="sc-bqWxrE iEntZY"
    >
      <div
        class="sc-ksBlkl RdDIT"
      >
        <ul>
          <li
            role="sign-in"
          >
            로그인
          </li>
          <li
            role="sign-up"
          >
            회원가입
          </li>
        </ul>
      </div>
      <div
        class="sc-hLBbgP FbSkc"
      >
        <form
          class="sc-eDvSVe juBFCv"
        >
          <div
            class="sc-jSUZER jhgDYF"
          >
            <input
              name="email"
              placeholder="이메일 입력하세요"
            />
            <span>
              이메일
            </span>
          </div>
          <div
            class="sc-jSUZER jhgDYF"
          >
            <input
              name="password"
              placeholder="비밀번호를 입력하세요"
              type="password"
            />
            <span>
              비밀번호
            </span>
          </div>
          <button>
            로그인
          </button>
          <div
            class="sc-gKPRtg eEOnMt"
          >
            아직 모가코 회원이 아니닌가요? 
            <span>
              모가코와 함께하기
            </span>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
